src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -661,11 +657,22 @@ private static TypeWithAnnotations DelegateReturnTypeWithAnnotations(MethodSymbo
                 refKind: default,
                 returnType: default);
 
+            ArrayBuilder<ConstantValue?>? parameterDefaultValueBuilder = null;
             foreach (var param in lambdaSymbol.Parameters)
             {
-                parameterDefaultValueBuilder.Add(param.ExplicitDefaultConstantValue);
+                var constVal = param.ExplicitDefaultConstantValue;
+                if (constVal != null)
+                {
+                    if (parameterDefaultValueBuilder == null)
+                    {
+                        parameterDefaultValueBuilder = ArrayBuilder<ConstantValue?>.GetInstance(ParameterCount);
+                    }
+
+                    parameterDefaultValueBuilder.Add(param.ExplicitDefaultConstantValue);
```

Comment:
This isn't the correct location. We'll need to start by filling the array up to the current parameter with nulls.

Reply:
Ah good catch. And front-filling up to the current parameter should be sufficient since at this point we should never have a case where the array needs to look like `{null, ..., <value>, ..., null}` as that would imply that default params came before required ones

Reply:
I realize I don't have any test cases for this code yet because I hadn't gotten to the point in code yet where we would actually use this array